test/libsolidity/semanticTests/array/slices/array_slice_calldata_to_storage.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    int[][] s;
+    function f(int[] calldata b, uint256 start, uint256 end) public returns (int) {
+        s.push();
+        s[0] = int[](b[start:end]);
```

Comment:
Without explicitly converting to `int[]` this would not pass `TypeChecker`. That makes this first converting to calldata array, than copies to storage, which is done trough already implemented code of array conversion.

Does it makes sense to allow implicit conversion of slices to arrays as I think it will work just fine? (Of course, this should be done as a breaking change)

Reply:
We discussed in channel, that this PR should allow conversions as we don't see any problem with it.

If we end up with not allowing this implicit conversion, we agreed that, also the weird inline case should be disallowed as well.